pkg/applicationserver/io/packages/alcsync/v1/messages.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alcsyncv1
+
+import "encoding/binary"
+
+// AppTimeReq is the device request for time correction.
+type AppTimeReq struct {
+	// DeviceTime represents the device GPS time in seconds.
+	DeviceTime  uint32
```

Comment:
Maybe you will have a bit of an easier time if you use `time.Time` here and marshal/unmarshal do the conversion to `uint32`.